### PR TITLE
travis: enable code coverage test using codecov

### PR DESCRIPTION
**Pull request by lamxTyler**
### What problem does this PR solve? 

Enable the code coverage test in travis ci.

### What is changed and how it works?

Enable the code coverage in the config file and upload the result to codecov.

### Check List 

Tests 

 - No code

Code changes

 - None

Side effects

 - None

Related changes

 - None

---
This change is [Reviewable](https://reviewable.io/reviews/pingcap/tidb/8910)

### CODE REVIEW

**Comment by winoros** [comment]
Cloud we use this integration directly?
https://github.com/marketplace/codecov/plan/MDIyOk1hcmtldHBsYWNlTGlzdGluZ1BsYW4xNg==#pricing-and-setup

**Comment by lamxTyler** [comment]
@winoros I thinks it is what this pr is doing. We need to upload coverage reports from the CI builds.

**Comment by winoros** [comment]
Okay got it :joy:

File: .travis.yml [hunk 1]
```diff
@@ -7,6 +7,14 @@ go:
 
 env:
     - TRAVIS_COVERAGE=0
+    - TRAVIS_COVERAGE=1
+
+# Run coverage tests.
+matrix:
+  fast_finish: true
+  allow_failures:
+    - go: "1.11.2"
```
Comment (shenli) [inline .travis.yml]: @tiancaiamao Do we need golang 1.11.4?

**Comment by winoros** [comment]
When will the robot `codecov-io` report the status?

**Comment by lamxTyler** [comment]
@winoros When this PR gets merged.

**Comment by tiancaiamao** [comment]
LGTM

**Comment by winoros** [comment]
@lamxTyler Oh i mean when a new pr is pushed and updated later, when will the robot report the coverage status?

**Pull request merged**